lib/serializer/cbor/aws_iot_serializer_tinycbor_decoder.c

```diff
@@ -23,382 +23,453 @@
-                .signedInt = i
-            };
-            break;
-        }
-
-        case AWS_IOT_SERIALIZER_SCALAR_TEXT_STRING:
-        {
-            size_t length;
-
-
-            /* user doesn't provide the buffer */
-            if (pValueObject->value.pString == NULL)
-            {
-
-                if( cbor_value_is_length_known( pCborValue ) )
-                {
-                    /*
-                     * If the length of the CBOR string is known, then its
-                     * a finite length string. We store the pointer to the start of the string
-                     * and length as the length of the string.
-                     *
-                     */
-                    cborError = cbor_value_get_string_length( pCborValue, &length );
-                    if( _noCborError( cborError ) )
-                    {
-                        pValueObject->value.pString =  ( uint8_t* ) cbor_value_get_next_byte( pCborValue ) - length;
-                        pValueObject->value.stringLength = length;
-                    }
-                    else
-                    {
-                        returnError = AWS_IOT_SERIALIZER_INTERNAL_FAILURE;
-                    }
-                }
-                else
-                {
-                    /*
-                     * This is an infinite length string.
-                     * Calculate and save required length to pValueObject, leave the pString pointer
-                     * as null so user can allocate the required buffer.
-                     */
-                    cbor_value_calculate_string_length(pCborValue, &length);
-                    pValueObject->value.stringLength = length;
-                    returnError = AWS_IOT_SERIALIZER_BUFFER_TOO_SMALL;
-                }
-            }
-            else /* user provides the buffer */
-            {
-                /* copy the string to the buffer */
-                cbor_value_copy_text_string(pCborValue, ( char * ) pValueObject->value.pString, &length, NULL);
-                pValueObject->value.stringLength = length;
-            }
-
-            break;
-        }
-
-        case AWS_IOT_SERIALIZER_CONTAINER_MAP:
+        /* Save to decoder object's handle. */
+        pDecoderObject->pHandle = ( void * ) pCborValueWrapper;
+    }
+    else /* Create scalar object. */
+    {
+        switch( dataType )
         {
-            pValueObject->pHandle = pCborDecoder;
-            break;
-        }
-
-        default:
-            returnError = AWS_IOT_SERIALIZER_UNDEFINED_TYPE;
+            case AWS_IOT_SERIALIZER_SCALAR_SIGNED_INT:
+               {
+                   int i = 0;
+                   cborError = cbor_value_get_int( pCborValue, &i );
+
+                   /* TODO: assert no error on _createDecoderObject */
+
+                   pDecoderObject->value = ( AwsIotSerializerScalarValue_t )
+                   {
+                       .signedInt = i
```

Comment:
We could just do - pDecoderObject->value.signedInt = i

Reply:
Yes, that is cleaner